### ISSUE TITLE
y.sh cargo run swallows the signal emitted by the running process

### ISSUE THREAD

**Issue by antoyo** [rust-lang/rustc_codegen_gcc#496]
For instance, if the process segfaults, `./y.sh cargo run` will not emit a SIGSEGV signal while it should do that.

**Comment by GuillaumeGomez** [comment]
I don't think signals should be emitted by the parent as well (in this case, SIGSEGV tells that a memory access was wrong, which is only the case of the child, not the parent). We should display which signal ended the process though if any.

**Comment by antoyo** [comment]
`cargo` does it, so I believe we should do the same.

**Comment by bjorn3** [comment]
On Unix you could exec cargo instead of spawning it as subprocess. That is what cargo-clif and rustc-clif do.

**Comment by antoyo** [comment]
In my sync PR, it seems like this is still happening, re-opening just in case.

**Comment by antoyo** [comment]
So, it looks like we return `1` instead of raising the signal of the child process.

**Comment by GuillaumeGomez** [comment]
Do we want to re-raise the signal though?

**Comment by antoyo** [comment]
Yes. I'd like the same behavior as `cargo` for the `y.sh cargo` command.
